table/namespace_classifier.go

```diff
@@ -144,7 +144,7 @@ func (c *tableNamespaceClassifier) GetRegionNamespace(regionInfo *core.RegionInf
 	c.RLock()
 	defer c.RUnlock()
 
-	tableID := c.getTableID(regionInfo)
+	tableID := c.tableIDDecoder.DecodeTableID(regionInfo.StartKey)
```

Comment:
@disksing 

Do we need to set metadata region to the special namespace later?

Reply:
Good idea.

Reply:
File an issue? @disksing